nntrainer/graph/network_graph.cpp

```diff
@@ -226,8 +230,15 @@ void NetworkGraph::markNodesForBackwarding() {
         lnode->needsCalcDerivative(true);
       }
 #endif
-      for (auto const &out_layer : lnode->getOutputLayers()) {
-        must_support_backwarding.insert(out_layer);
+
+      for (auto i = 0u, num_node = lnode->getNumOutputConnections();
+           i < num_node; ++i) {
+        auto conn = lnode->getOutputConnection(i);
+        if (!conn) {
```

Comment:
what does an empty output connection signify?

Reply:
When multiout, it is possible scenario that connection is empty.
```ini
[some layer]

[a]
input_layers=some_layer(0)

[b]
input_layers=some_layer(3)
```

But I am not defining this as a bug because below will be valid equivalent in pytorch (issuing warning at different spot)

```python
a, _, b = some_layers(in)
```

Reply:
fine for now I guess. This is worst case some unnecessary computation.

Reply:
Yeah, let's make a proper discussion later :)